modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -0,0 +1,539 @@
+
+        if (seg == null) {
+            fileWriter.logSmallBufferMessage();
+
+            return;
+        }
+
+        // Ring buffer closed (writer stopping) or maximum size reached.
+        if (seg.buffer() == null) {
+            seg.release();
+
+            if (!fileWriter.isCancelled())
+                fileWriter.onMaxFileSizeReached();
+
+            return;
+        }
+
+        ByteBuffer buf = seg.buffer();
+
+        buf.put((byte)op.ordinal());
+
+        writer.accept(buf);
+
+        seg.release();
+    }
+
+    /** @return {@code True} if string is cached. {@code False} if need write string.  */
+    private boolean stringCached(String str) {
+        boolean cached = cachedStrings.contains(str.hashCode());
+
+        if (!cached)
+            cachedStrings.add(str.hashCode());
+
+        return cached;
+    }
+
+    /** @return Performance statistics file. */
+    private static File statisticsFile(GridKernalContext ctx) throws IgniteCheckedException {
+        String igniteWorkDir = U.workDirectory(ctx.config().getWorkDirectory(), ctx.config().getIgniteHome());
+
+        File fileDir = U.resolveWorkDirectory(igniteWorkDir, PERFORMANCE_STAT_DIR, false);
+
+        return new File(fileDir, "node-" + ctx.localNodeId() + ".prf");
+    }
+
+    /** Writes {@link UUID} to buffer. */
+    private static void writeUuid(ByteBuffer buf, UUID uuid) {
+        buf.putLong(uuid.getMostSignificantBits());
+        buf.putLong(uuid.getLeastSignificantBits());
+    }
+
+    /** Writes {@link IgniteUuid} to buffer. */
+    private static void writeIgniteUuid(ByteBuffer buf, IgniteUuid uuid) {
+        buf.putLong(uuid.globalId().getMostSignificantBits());
+        buf.putLong(uuid.globalId().getLeastSignificantBits());
+        buf.putLong(uuid.localId());
+    }
+
+    /** @return {@code True} if collecting performance statistics enabled. */
+    public boolean enabled() {
+        return enabled;
+    }
+
+    /** Worker to write to performance statistics file. */
+    private class FileWriter extends GridWorker {
+        /** Performance statistics file I/O. */
+        private final FileIO fileIo;
+
+        /** File write buffer. */
+        private final SegmentedRingByteBuffer ringByteBuffer;
+
+        /** Minimal batch size to flush in bytes. */
+        private final int flushBatchSize;
+
+        /** Size of ready for flushing bytes. */
+        private final AtomicInteger readyForFlushSize = new AtomicInteger();
+
+        /** {@code True} if the small buffer warning message logged. */
+        private final AtomicBoolean smallBufLogged = new AtomicBoolean();
+
+        /** {@code True} if worker stopped due to maximum file size reached. */
+        private final AtomicBoolean stopByMaxSize = new AtomicBoolean();
+
+        /**
+         * @param ctx Kernal context.
+         * @param fileIo Performance statistics file I/O.
+         * @param maxFileSize Maximum file size in bytes.
```

Comment:
Let's remove those 3 parameters, because we only use constants for it, for now.

Reply:
Done.